lib/route_types.txt

```diff
@@ -78,6 +78,7 @@ ZEBRA_ROUTE_BGP_DIRECT, bgp-direct, NULL,  'b', 0, 0, "BGP-Direct"
 ZEBRA_ROUTE_BGP_DIRECT_EXT, bgp-direct-to-nve-groups, NULL, 'e', 0, 0, "BGP2VNC"
 ZEBRA_ROUTE_BABEL,      babel,     babeld, 'A', 1, 1, "Babel"
 ZEBRA_ROUTE_SHARP,    sharp,     sharpd, 'D', 1, 1, "SHARP"
+ZEBRA_ROUTE_OSPF_SR,    ospf-sr,   ospfd,  's', 1, 0, "OSPF-SR" 
```

Comment:
Can you help me to understand the reasoning on the addition of a new route-type here?

Reply:
@rwestphal @mwinter-osr @riw777 do you have opinions about the addition of a new route-type.  I believe Olivier was trying to allow distinctions in 'show ip route' so you can tell at a glance if a route is MPLS or not.  But I believe this would easily confuse people in that they have to now do 'redistribute ospf-sr' if they want to redistribute an ospf mpls route.

Reply:
I would add that I'm not sure that it is safe to redistribute an OSPF-SR route in BGP. I mean redistribute an MPLS route with a specific label into BGP.

Reply:
Why wouldn't it be safe to redistribute an SR OSPF route into BGP? I'm trying to think of a situation where this would be a problem, given the label stack is carried with the route in redistribution. Whether or not we'd want SR routes to be redistributed by default is a different question -- I'm trying to think of which would be more surprising, redistributing OSPF and having the SR routes pulled as well, or redistributing OSPF and finding that not all the routes are bring redistributed until I do another kind of redistribution... My guess is the latter, rather than the former.

Given redistribution is okay/workable, then I'd say not to put the new route type in...

Reply:
> given the label stack is carried with the route in redistribution

That's not the case as of now. zebra sends redistributed routes without their labels to the client daemons. For LDP-labeled routes that's not a problem, BGP and the other routing daemons shouldn't care if a route is LDP-labeled or not. What matters is the reachability information only.

For SR-labeled routes the situation might be different, I don't know. IMO the safest thing to do is to allow the redistribution of OSPF-SR routes, but without their labels. This way reachability is preserved but the SR tunnels start only where intended. Mixing OSPF/IS-IS/BGP SR domains might be doable but I believe that it should involve more than simple route redistribution to make it happen.

Reply:
In fact, standard OSPF routes still be there. With Segment Routing, you just add labelled OSPF route to Prefix advertised by the other OSPF routers. For example, an extract of `sh ip route 192.168.1.9/32` gives: `` dcn3# sh ip route 192.168.1.9/32
Routing entry for 192.168.1.9/32
  Known via "ospf-sr", distance 0, metric 1, best
  Last update 00:02:59 ago
  * 192.168.0.65, via sw_ml, label 20900

Routing entry for 192.168.1.9/32
  Known via "ospf", distance 110, metric 12
  Last update 00:03:00 ago
    192.168.0.65, via sw_ml
``
Both routes are there. The SR is install as in the code I put a higher priority (distance and metric are not pertinent). In case of route redistribution, OSPF routes will be redistribute, not SR one (I hope). If I don't made any distinction between both, what's happen ? Is the SR route will continue to be selected as the `best` one ? Is BGP, IS-IS ... will don't care about the label ? If answer are yes for both, I agree that new route type is not necessary and to remove the OSPF-SR route type from the code

Reply:
Yes for both from my perspective :) Also, other vendors don't seem to be creating new route types for SR routes. I'm not saying we should blindly follow their lead, but it's a point to consider.